genesis/src/lib.rs

```diff
@@ -28,16 +31,121 @@ pub static GENESIS_ACCOUNT_ID: Lazy<AccountId> =
     Lazy::new(|| AccountId::new(GENESIS_DOMAIN_ID.clone(), "genesis".parse().expect("Valid")));
 
 /// Genesis transaction
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Decode, Encode)]
 #[repr(transparent)]
 pub struct GenesisTransaction(pub SignedTransaction);
 
+impl GenesisTransaction {
+    /// Construct single genesis transaction with [`Executor`] and transactions unified under a single transaction
+    pub fn new_unified(
+        raw_block: RawGenesisBlock,
+        chain_id: &ChainId,
+        genesis_key_pair: &KeyPair,
+    ) -> GenesisTransaction {
+        Self(raw_block.unify().sign(chain_id.clone(), genesis_key_pair))
+    }
+}
+
+/// Different errors as a result of parsing hex string for `GenesisSignature`
+#[derive(Copy, Clone, Debug, displaydoc::Display)]
+pub enum GenesisSignatureParseError {
+    ///  Hex string could not be decoded into scale encoded bytes
+    HexDecodeError,
+    ///  The config has incorrect scale format and cannot be decoded
+    ScaleDecodeError,
+}
+
+impl std::error::Error for GenesisSignatureParseError {}
+
+/// [`SignedGenesisConfig`] contains data that is used for loading signed genesis from config.
+#[derive(Debug, Clone, PartialEq, Eq, Decode, Encode)]
+pub struct GenesisSignature {
```

Comment:
As a small note, `iroha_genesis::GenesisSignature` contains name duplication, and may be shortened to `iroha_genesis::Signature`. Up to your preference here.

Same applies to `GenesisSignatureParseError`, `GenesisNetwork` etc.

Reply:
I thought that we already have lots of "signatures" and decided to distinguish them like that.
And yeah, we had `GenesisNetwork` for a long time so it's kinda similar in that sense, or so I thought.